core/arch/emit_utils_shared.c

```diff
@@ -5120,7 +5120,11 @@ update_syscalls(dcontext_t *dcontext)
     generated_code_t *code = THREAD_GENCODE(dcontext);
     protect_generated_code(code, WRITABLE);
     pc = get_do_syscall_entry(dcontext);
-    update_syscall(dcontext, pc);
+    if (IF_X64_ELSE(true,
+                    !(get_syscall_method() == SYSCALL_METHOD_SYSCALL &&
```

Comment:
So the proposal is to have the global syscalls remain `int 0x80`, rather than figuring out what the problem is with 32-bit `syscall`?  Please at least include a comment explaining this, that the actual problem is still unknown.

But what about:
+ The `syscall` instruction clobbers `ecx` IIRC -- so wouldn't the registers holding parameters be different?  Is the 2nd param not in ecx but somewhere else??
+ What about inlined syscalls inside a fragment in the code cache?  They will be left as `syscall`.  Does the same problem that affects the global generated code affect them, and thus this is not a complete solution?